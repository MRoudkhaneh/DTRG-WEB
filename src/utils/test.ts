const shallowFinder = (component, attr) => component.find(`[id='${attr}']`);

export { shallowFinder };
