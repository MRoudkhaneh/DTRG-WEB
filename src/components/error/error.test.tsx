import { Error } from ".";
import { idFinder, useTree, setUp } from "test";

describe("Error", () => {
  let component;
  beforeEach(() => (component = setUp(Error, { error: "test" })));
  it("SHOULD MATCH SNAPSHOT", () => {
    let error = useTree(Error);
    expect(error).toMatchSnapshot();
  });
  it("SHOULD RENDER WRAPPER", () => {
    const wrapper = idFinder(component, "error");
    expect(wrapper).toHaveLength(1);
  });
  it("SHOULD RENDER NOTHING WHEN PROPS IS UNDEFINED", () => {
    const wrapper = idFinder(setUp(Error, {}), "error");
    expect(wrapper).toHaveLength(0);
  });
});
