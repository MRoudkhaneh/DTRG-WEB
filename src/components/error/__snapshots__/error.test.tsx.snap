// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error SHOULD MATCH SNAPSHOT 1`] = `null`;
